### ISSUE TITLE
Parsing in `pathWithoutExt(filePath)` does not work for Windows paths

### ISSUE THREAD

**Issue by thewalla07** [eslint/eslint#12507]
Environment is Windows 10 Version 1809

Environment Info:

Node version: v10.15.0
npm version: v6.4.1
Local ESLint version: v6.5.1 (Currently used)
Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?** default (N/A)

**Please show your full configuration:** (N/A)

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

We run ESLint in Jenkins CI on a Windows host, and get the following type of output from failures
![image](https://user-images.githubusercontent.com/9087218/67778911-cffdde80-fa5b-11e9-8baa-e5e2e633040e.png)

Currently, this means we do not know the filepath for the files with errors, only for the ones which passed.

This issue arises as a result of PR #11683. See below:

https://github.com/eslint/eslint/blob/287ca562811d037bde09a47af7f5b9c7b741e022/lib/cli-engine/formatters/junit.js#L28-L36

The above code assumes that POSIX filePaths are used, meaning that the function does not work as expected in Windows.

```bash
npx eslint home_page.js --format junit
```

**What did you expect to happen?**
I expect that in the `classname` tag below I can see the full path to the file, without the file extension for visualization within Jenkins CI on a Windows host. i.e. `C:\path\to\file` 
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
<testsuite package="org.eslint" time="0" tests="1" errors="1" name="C:\git\projectdirectory\home_page.js">
<testcase time="0" name="org.eslint.prefer-const" classname="C:\git\projectdirectory\home_page"><failure message="&apos;msg&apos; is never reassigned. Use &apos;const&apos; instead."><![CDATA[line 7, col 17, Error - &apos;msg&apos; is never reassigned. Use &apos;const&apos; instead. (prefer-const)]]></failure></testcase>
</testsuite>
</testsuites>
```

**What actually happened? Please include the actual, raw output from ESLint.**
The function above parses `filePath` incorrectly on Windows. `path.posix.dirname('C:\\path\\to\\file.js')` resolves to `.` on Windows, as `posix` forces an assumption of POSIX style paths.

This results in the following XML, without the full path to the file in the `classname` attribute:
```xml
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
<testsuite package="org.eslint" time="0" tests="1" errors="1" name="C:\git\projectdirectory\home_page.js">
<testcase time="0" name="org.eslint.prefer-const" classname="home_page"><failure message="&apos;msg&apos; is never reassigned. Use &apos;const&apos; instead."><![CDATA[line 7, col 17, Error - &apos;msg&apos; is never reassigned. Use &apos;const&apos; instead. (prefer-const)]]></failure></testcase>
</testsuite>
</testsuites>
```

**Are you willing to submit a pull request to fix this bug?**
Yes.

I would propose a fix where we do not use the `posix` specific implementation of Node's `path` library, and we let Node decide which implementation to use based on the underlying OS.

The main question arises in testing. In https://github.com/eslint/eslint/blob/master/tests/lib/cli-engine/formatters/junit.js is it possible to run a test with both POSIX and Windows style paths? My suggestion would be to have the test cases adapt to the current OS, but opinions would be valued here.

**Comment by thewalla07** [comment]
Working on this, created PR to fix this issue

**Comment by thewalla07** [comment]
@kaicataldo any chance this will be reopened, or should the associated PR be abandoned?

**Comment by kaicataldo** [comment]
Thanks for the ping. Let me see if I can get some eyes on this PR!
